mlflow/sklearn/__init__.py

```diff
@@ -720,7 +726,30 @@ def _log_posttraining_metadata(estimator, *args, **kwargs):
             else:
                 try_mlflow_log(mlflow.log_metric, "training_score", training_score)
 
-        try_mlflow_log(log_model, estimator, artifact_path="model")
+        input_example = None
+        signature = None
+        if hasattr(estimator, "predict"):
+            try:
+                # Fetch an input example using the first several rows of the array-like
+                # training data supplied to the training routine (e.g., `fit()`)
+                SAMPLE_ROWS = 5
+                fit_arg_names = _get_arg_names(estimator.fit)
+                X_var_name, y_var_name = fit_arg_names[:2]
+                input_example = _get_Xy(args, kwargs, X_var_name, y_var_name)[0][:SAMPLE_ROWS]
+
+                model_output = estimator.predict(input_example)
+                signature = infer_signature(input_example, model_output)
+            except Exception as e:
+                input_example = None
```

Comment:
Added this line to prevent logging `input_example` when  `estimator.predict` or `infer_signature` fails.